docs/linux/concepts/concurrency.md

```diff
@@ -53,29 +65,38 @@ struct {
 	__type(key, int);
 	__type(value, struct concurrent_element);
 	__uint(max_entries, 100);
-} SEC(".maps") concurrent_map ;
+} concurrent_map SEC(".maps");
 ```
 
-Then in your code, you can take the lock with [`bpf_spin_lock`](../helper-function/bpf_spin_lock.md), do whatever you need to do, and release the lock with [`bpf_spin_unlock`](../helper-function/bpf_spin_unlock.md). 
+Then in your code, you can take the lock with [`bpf_spin_lock`](../helper-function/bpf_spin_lock.md), do whatever you need to do, and release the lock with [`bpf_spin_unlock`](../helper-function/bpf_spin_unlock.md). In this example, we simply increment the number of times the `sys_enter` tracepoint is called.
 
 ```c
-int bpf_program(struct pt_regs *ctx) {
+SEC("tp_btf/sys_enter")
+int sys_enter_count(void *ctx) {
 	int key = 0;
 	struct concurrent_element init_value = {};
 	struct concurrent_element *read_value;
-
-	bpf_map_create_elem(&concurrent_map, &key, &init_value, BPF_NOEXIST);
+	bpf_map_update_elem(&concurrent_map, &key, &init_value, BPF_NOEXIST);
 
 	read_value = bpf_map_lookup_elem(&concurrent_map, &key);
+	if(!read_value)
+	{
+		return 0;
+	}
+
 	bpf_spin_lock(&read_value->semaphore);
-	read_value->count += 100;
+	read_value->count += 1;
 	bpf_spin_unlock(&read_value->semaphore);
+	return 0;
 }
 ```
 
 !!! warning
 	The [verifier](linux-verifier.md) will fail if there exists a code path where you take a lock and never release it. You are also not to take more than one lock at a time since that can cause a [deadlock](https://en.wikipedia.org/wiki/Deadlock) scenario.
 
+!!! warning
+	Not all BPF program types support `bpf_spin_lock` so be sure to check your program is supported if you face verifier errors. The verifier, in 6.8.0, performs [these checks](https://github.com/torvalds/linux/blob/e8f897f4afef0031fe618a8e94127a0934896aba/kernel/bpf/verifier.c#L17882) but since they can change across kernel versions be sure to check what is allowed by your kernel version.
+
```

Comment:
Great warning, though I would rater link to the "Program types" list on the `bpf_spin_lock` page, and make sure that list is correct.

Reply:
oh sorry, I didn't notice the `bpf_spin_lock` specific page, sure I will do that!

Reply:
done! The current list seems correct!